Insrt-Finance/insrt-v2-contracts
# Feat/TokenBridge implementation (#240)

* feat: main implementation, scripts & tests

* Add: address(this) as minter in scripts

* Add tests and fix scripts

* scripts working

* Fix: distributionFractionBP 0 check

* Add zero check in setDistributionFractionBP

* Fix: general styling

* Fix: Styling completely fixed

* refactor: tests

* Feat: execute test working

* Add: security check and last tests

* Add: secrets

* remove secrets

* Add: .gitignore

* Fix: CI file

* new CI

* new trial

* newer trial

* Remove temp file

* Fix: CI

* Update contracts/facets/Token/TokenBridgeInternal.sol



* Fix: natspec

* Add: test for distribution fraction BP

* fix: editor formatter

* Fix: execute test

* Fix: repristinate original CI

* Update contracts/facets/Token/TokenBridgeInternal.sol



* Update contracts/facets/Token/TokenBridgeInternal.sol



* Update test/facets/Token/behaviors/bridgeToken.t.sol



* Add: execute comment

* Fix: add return value names

* Update test/facets/Token/behaviors/bridgeToken.t.sol



* Rename: getDestinationAddress

* Remove: payable comment

* Fix:deploy script

* Remove: error from tests

* Fix: add constant for bridge amount

* Add: Test with expect call to bridgeToken function

* Fix: tests styling

* Fix: semantic styling in tests

* Add: test to setup in bridgeToken

* Add: supported addresses length

* Refactor: batchEnables and tests

* Add: all tests for new functions

* Fix: executable inherited from external contract

* Fix: mint entire amount when bridging

* Fix: _execute style

* Fix: CI forge coverage

* Update contracts/facets/Token/ITokenBridge.sol



* Update contracts/facets/Token/ITokenBridge.sol



* Update contracts/facets/Token/ITokenBridgeInternal.sol



* Add: params in natspec for events

* Fix: make uint8 more explicit

* Fix: make immutable capital

* Fix: move utility function at the bottom of the file

* Fix: modify natspec related to address length enabling

* Update test/facets/Token/behaviors/batchDisableAddressLength.t.sol



* Update contracts/facets/Token/TokenBridgeInternal.sol



* Update contracts/facets/Token/TokenBridgeInternal.sol



* Update test/facets/Token/behaviors/getDestinationAddress.t.sol



* Update test/facets/Token/behaviors/execute.t.sol



* Update test/facets/Token/behaviors/execute.t.sol



* Update test/facets/Token/behaviors/execute.t.sol



* Update test/facets/Token/behaviors/execute.t.sol



* Update test/facets/Token/behaviors/bridgeToken.t.sol



* Update test/facets/Token/behaviors/batchDisableAddressLength.t.sol



* Fix: revert style

* fix: style revert

* Fix: typos

* Fix: token bridge natspec

* Fix: token script after deployment

* Add: basic asserts after setup

---------
